### DISCUSSION TITLE
RPC Queries work, health check doesn't.

### DISCUSSION THREAD

**Discussion by eccentricexit** [erigontech/erigon#3064] [Q&A]
Health check fails and the error message didn't really get me in any direction.

```
$ curl localhost:8545/health
{"check_block":"DISABLED","healthcheck_query":"ERROR: unexpected end of JSON input","min_peer_count":"DISABLED"}
```
RPC Daemon output:
`unable to process healthcheck request    error="unexpected end of JSON input"`

Interestingly, running a test call returns what looks like a proper result:

```
curl -X POST -H "Content-Type: application/json" --data '{"jsonrpc": "2.0", "method": "eth_blockNumber", "params": [], "id":1}' localhost:8545
{"jsonrpc":"2.0","id":1,"result":"0xd14cf2"}
```

Logs from my erigon node and the RPC node makes it look like they are talking correctly too:

![Screenshot from 2021-11-30 16-47-21](https://user-images.githubusercontent.com/9872474/144117250-4a729568-dd7b-4fb2-9d3a-0542362c91a3.png)

Any suggestions?

Erigon version:
```
$ ./erigon/build/bin/erigon --version
erigon version 2021.11.4-beta
```

**Comment by AskAlexSharov** [comment]
@mandrigin do you have time to take a look?

**Comment by blampe** [comment]
Also seeing this on `2021.12.2-beta`. It seems the endpoint only accepts a JSON-encoded POST body. I would expect it to respond to `GET`, optionally taking some parameters like `GET /health&min_peers=3`.

**Comment by daoleno** [comment]
[HealthCheck GET & POST](https://github.com/ledgerwatch/erigon/blob/devel/cmd/rpcdaemon/README.md#healthcheck)
